packages/forma-36-react-components/src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FunctionComponent, KeyboardEvent } from 'react';
+import FormLabel from '../Typography/Paragraph';
+import classNames from 'classnames';
+import styles from './Switch.css';
+
+export interface SwitchProps {
+  onToggle?: (isChecked: boolean) => void;
+  isChecked?: boolean;
+  isDisabled?: boolean;
+  labelText: string;
+  className?: string;
+}
+
+export const Switch: FunctionComponent<SwitchProps> = (props: SwitchProps) => {
+  return (
+    <div className={classNames(styles['Switch__wrapper'], props.className)}>
+      <input
+        type="checkbox"
+        onClick={onToggle}
+        onKeyUp={onKeyUp}
+        className={classNames(styles['Switch'], {
+          [styles['Switch--checked']]: props.isChecked,
+          [styles['Switch--disabled']]: props.isDisabled,
+        })}
+      />
+      <FormLabel
+        className={classNames(styles['Switch__label'], {
+          [styles['Switch__label--disabled']]: props.isDisabled,
+        })}
+      >
+        {props.labelText}
+      </FormLabel>
+    </div>
+  );
+
+  function onToggle() {
+    if (props.onToggle) {
+      props.onToggle(!props.isChecked);
+    }
+  }
+
+  function onKeyUp(e: KeyboardEvent<HTMLElement>) {
```

Comment:
I'm a bit worried with how we use the checkbox here. In this case enter would not set the checkbox to checked since natively it would be space which checks the input. But as we implement it now it does not matter since we're just returning the !props.isChecked which means the checkbox elements state and what it returns will get out of sync.. can this result in unexpected behaviour? So basically we're just using the checkbox to render some interactive element..

Reply:
Good catch, just updated as well as adding more snapshot tests